ads/google/csa.js

```diff
@@ -0,0 +1,349 @@
+  if (data['afshPageOptions'] != null) {
+    try {
+      afshPageOptions = JSON.parse(data['afshPageOptions']);
+      afshPageOptions['source'] = 'amp';
+      afshPageOptions['referer'] = window.context.referrer;
+    } catch (e) {}
+  }
+
+  // Parse AFSh adblock options
+  let afshAdblockOptions = {};
+  if (data['afshAdblockOptions'] != null) {
+    try {
+      afshAdblockOptions = JSON.parse(data['afshAdblockOptions']);
+
+      // Set container to the container we just created
+      afshAdblockOptions['container'] = containerId;
+
+      // Set the width to the width of the screen if necessary
+      if (afshAdblockOptions['width'] == 'auto') {
+        afshAdblockOptions['width'] = width;
+      }
+    } catch (e) {}
+  }
+
+  // Parse AFS page options
+  let afsPageOptions = {};
+  if (data['afsPageOptions'] != null) {
+    try {
+      afsPageOptions = JSON.parse(data['afsPageOptions']);
+      afsPageOptions['source'] = 'amp';
+      afsPageOptions['referrer'] = window.context.referrer;
+    } catch (e) {}
+  }
+
+  // Parse AFS adblock options
+  let afsAdblockOptions = {};
+  if (data['afsAdblockOptions'] != null) {
+    try {
+      afsAdblockOptions = JSON.parse(data['afsAdblockOptions']);
+
+      // Set the container to the container we just created
+      afsAdblockOptions['container'] = containerId;
+
+    } catch (e) {}
+  }
+
+  /* Time in ms to wait before executing orientation change function logic */
+  const orientationChangeTimeout = 250;
+
+  /**
+   * Resize the AMP iframe if the CSA container changes in size upon rotation.
+   * This is needed for an iOS bug found in versions 10.0.1 and below that
+   * doesn't properly reflow the iframe upon orientation change.
+   */
+  window.addEventListener('orientationchange', function() {
+
+    // Save the height of the container before the event listener triggers
+    const oldHeight = document.getElementById('csacontainer').style.height;
+
+    setTimeout(function() {
+
+      // Force DOM reflow and repaint
+      /*eslint-disable no-unused-vars*/
+      const ignore = document.body.offsetHeight;
+      /*eslint-enable no-unused-vars*/
+
+      // Capture new height
+      const container = document.getElementById('csacontainer');
+      let newHeight = container.style.height;
+
+      // In older versions of iOS, this height will be different because the
+      // container height is resized.
+      // In Chrome and iOS 10.0.2 the height is the same because
+      // the container isn't resized.
+      if (oldHeight != newHeight && newHeight != currentAmpHeight) {
+
+        // style.height returns "60px" (for example), so turn this into an int
+        newHeight = parseInt(newHeight, 10);
+
+        // Also update the onclick function to resize to the right height.
+        const overflow = document.getElementById('overflow');
+        if (overflow) {
+          overflow.onclick = function() {
+            window.context.requestResize('auto', newHeight);
```

Comment:
the API does not accept 'auto'. If you only want to change the height, just pass `undefined`